src/04-modality-specific-files/11-functional-near-infrared-spectroscopy.md

```diff
@@ -0,0 +1,400 @@
+# Functional Near-Infrared Spectroscopy
+
+Support for functional Near-Infrared Spectroscopy (fNIRS) was developed as a
+[BIDS Extension Proposal](../07-extensions.md#bids-extension-proposals).
+Please see [Citing BIDS](../01-introduction.md#citing-bids)
+on how to appropriately credit this extension when referring to it in the
+context of the academic literature.
+
+Several [example fNIRS datasets](https://github.com/bids-standard/bids-examples#fnirs-datasets)
+have been formatted using this specification and can be used for practical guidance when curating a new dataset.
+
+## fNIRS recording data
+
+{{ MACROS___make_filename_template(datatypes=["nirs"], suffixes=["nirs", "events", "channels", "optodes", "coordsystem"]) }}
+
+Only the Shared Near Infrared Spectroscopy Format ([SNIRF](https://github.com/fNIRS/snirf))
+file specification is supported in BIDS. The SNIRF
+specification supports one or more fNIRS datasets to be stored in a single
+`.snirf` file. However, to be BIDS compatible, each SNIRF file MUST contain
+only a single run. A limited set of fields from the SNIRF specification are
```

Comment:
There is no JS reader for SNIRF. And I know of no initiatives to achieve this.

However, the SNIRF files are just HDF5 files. And I found a javascript library called `jsfive` that claims to be "A pure javascript HDF5 file reader". So I think it would be feasible in principle. 

I am glad this is not a blocker. I am not versed in javascript, so this would be a very large undertaking for me to implement.

Reply:
sounds cool! Would you mind opening an issue on bids-validator that tracks exactly this feature for a future enhancement?

Reply:
https://github.com/bids-standard/bids-validator/issues/1472